### PR TITLE
[fast-double-parser] update to 0.8.0

### PR DESCRIPTION
**Pull request by jiayuehua**
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] SHA512s are updated for each updated download
- [ ] ~The "supports" clause reflects platforms that may be fixed by this new version~
- [ ] ~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~
- [ ] ~Any patches that are no longer applied are deleted from the port's directory.~
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Review by Cheney-W — DISMISSED** [review]

**Comment by Osyotr** [comment]
The library now installs cmake config, so the port should use `vcpkg_cmake_*` functions  instead of just copying header file.

File: ports/fast-double-parser/portfile.cmake [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS
+        -DBUILD_TESTING=OFF
+)
 
-file(INSTALL "${SOURCE_PATH}/include/" DESTINATION "${CURRENT_PACKAGES_DIR}/include")
+vcpkg_cmake_install()
 
+#vcpkg_cmake_config_fixup(PACKAGE_NAME fast_double_parser CONFIG_PATH "lib/cmake/cpp-sort")
```
Comment (BillyONeal) [inline ports/fast-double-parser/portfile.cmake:19]: This is still commented out, it seems this is still being worked on?

Comment (jiayuehua) [reply ports/fast-double-parser/portfile.cmake]: this comment is not needed . I just Have deleted this comment .

File: ports/fast-double-parser/vcpkg.json [hunk 2]
```diff
@@ -1,7 +1,17 @@
...
-  "license": "Apache-2.0"
+  "license": "Apache-2.0",
+  "dependencies": [
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    },
+    {
+      "name": "vcpkg-cmake-config",
+      "host": true
```
Comment (Cheney-W) [inline ports/fast-double-parser/vcpkg.json:14]:
```suggestion
```
This dependency could be removed.

**Review by Cheney-W — CHANGES_REQUESTED** [review]

File: ports/fast-double-parser/vcpkg.json (hunk 2 above)
Comment (jiayuehua) [reply ports/fast-double-parser/vcpkg.json]: have just deleted

**Comment by dg0yt** [comment]
IMO this should be handled with `vcpkg_cmake_config_fixup(PACKAGE_NAME fast_double_parser)`:
~~~
fast-double-parser:x64-linux:/share/fast_double_parser/fast_double_parser-config.cmake
fast-double-parser:x64-linux:/share/fast_double_parser/fast_double_parser-targets.cmake

**Comment by Cheney-W** [comment]
@jiayuehua 
Could you please follow dg0yt's suggestion to add a call to the `vcpkg_cmake_config_fixup` function in the `portfile.cmake` file? At the same time, you need to add the `vcpkg-cmake-config` dependency in the `vcpkg.json` file.  
Although calling this function will not change the current output files or layout of this port, doing so can help prevent potential issues in the future if the upstream modifies the exported contents.

**Comment by Osyotr** [comment]
> Although calling this function will not change the current output files or layout of this port

This is because of the spurious `set(VCPKG_BUILD_TYPE release)` in the portfile.

**Comment by jiayuehua** [comment]
> @jiayuehua 
> 
> Could you please follow dg0yt's suggestion to add a call to the `vcpkg_cmake_config_fixup` function in the `portfile.cmake` file? At the same time, you need to add the `vcpkg-cmake-config` dependency in the `vcpkg.json` file.  
> 
> Although calling this function will not change the current output files or layout of this port, doing so can help prevent potential issues in the future if the upstream modifies the exported contents.

ok I will follow dg0yt's advice summit the change later.

**Comment by BillyONeal** [comment]
> > Although calling this function will not change the current output files or layout of this port
> 
> This is because of the spurious `set(VCPKG_BUILD_TYPE release)` in the portfile.

How is it "spurious"? Is the library not header only any longer?

**Review by BillyONeal — APPROVED** [review]

**Comment by Osyotr** [comment]
> How is it "spurious"? Is the library not header only any longer?

I was thinking about interface properties in debug config, but now I see that `set(VCPKG_BUILD_TYPE release)` + `vcpkg_cmake_config_fixup`  is widely used in vcpkg.
https://github.com/search?q=repo%3Amicrosoft%2Fvcpkg+set%28VCPKG_BUILD_TYPE+vcpkg_cmake_config_fixup&type=code

**Review by Osyotr — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**